### PR TITLE
fix some duplicated testgrid entries, update unit test

### PR DESCRIPTION
**Pull request by krzyzacy**
fix up something I found from https://github.com/kubernetes/test-infra/pull/9067
/shrug
/assign @michelle192837

### CODE REVIEW

File: testgrid/cmd/configurator/config_test.go [hunk 1]
```diff
@@ -190,10 +195,17 @@ func TestConfig(t *testing.T) {
 			}
 
 			// All dashboardtab within a dashboard must not have duplicated names
-			if dashboardtabmap[dashboardtab.Name] {
-				t.Errorf("Duplicated dashboardtab: %v", dashboardtab.Name)
+			if dashboardtabSet.Has(dashboardtab.Name) {
+				t.Errorf("Duplicated name in dashboard %s: %v", dashboard.Name, dashboardtab.Name)
+			} else {
```
Comment (mithrav) [inline testgrid/cmd/configurator/config_test.go]: Do you need the else clause here?

File: testgrid/cmd/configurator/config_test.go [hunk 2]
```diff
@@ -190,10 +195,17 @@ func TestConfig(t *testing.T) {
...
+				dashboardtabSet.Insert(dashboardtab.Name)
+			}
+
+			// All dashboardtab within a dashboard must not have duplicated testgroupnames
+			if dashboardtestgroupSet.Has(dashboardtab.TestGroupName) {
+				t.Errorf("Duplicated testgroupnames in dashboard %s: %v", dashboard.Name, dashboardtab.TestGroupName)
 			} else {
```
Comment (mithrav) [inline testgrid/cmd/configurator/config_test.go]: Likewise

File: testgrid/cmd/configurator/config_test.go (hunk 1 above)
Comment (krzyzacy) [reply testgrid/cmd/configurator/config_test.go]: https://golang.org/pkg/testing/#B.Errorf

errorf marks the test as fail, but continues execution :-)

**Pull request merged**